src/ServiceControl.Monitoring/QueueLength/TagExtensions.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Linq;
+
+using System.Collections.Generic;
+
+static class TagExtensions
+{
+    public static string GetTagValue(this IEnumerable<string> tags, string key)
+    {
+        string result;
+        if (!TryGetTagValue(tags, key, out result))
+        {
+            throw new Exception($"Tag {key} not found.");
+        }
+        return result;
+    }
+
+    public static bool TryGetTagValue(this IEnumerable<string> tags, string key, out string value)
+    {
+        value = tags
+            .Where(t => t.StartsWith($"{key}:", StringComparison.OrdinalIgnoreCase))
```

Comment:
Should we consider writing functions for all the key values? Like

`TryGetTypeTagValue` and have strings consts?